packages/js/src/generators/library/library.ts

```diff
@@ -471,25 +471,25 @@ export async function addLint(
           ruleSeverity = value;
           ruleOptions = {};
         }
-        if (options.bundler === 'vite' || options.unitTestRunner === 'vitest') {
-          ruleOptions.ignoredFiles ??= [];
-          ruleOptions.ignoredFiles.push(
-            '{projectRoot}/vite.config.{js,ts,mjs,mts}'
-          );
-          o.rules['@nx/dependency-checks'] = [ruleSeverity, ruleOptions];
+        const ignoredFiles = new Set(ruleOptions.ignoredFiles ?? []);
```

Comment:
Technically you could remove the `else if` from 477 and achieve the same result I think.

```js
if (options.bundler === "vite" || options.unitTestRunner === 'vitest') {
  ignoredFiles.add("{projectRoot}/vite.config.{js,ts,mjs,mts}");
}

if (options.bundler === "rollup") {
  ignoredFiles.add("{projectRoot}/rollup.config.{js,ts,mjs,mts,cjs,cts}");
} else if (options.bundler === "esbuild") {
  ignoredFiles.add("{projectRoot}/esbuild.config.{js,ts,mjs,mts}");
}

```

Reply:
It works, but I don't think it reads as well because you already check for `bundler === vite` first, so, there's no way it can be `bundler === rollup` if that was the case. Checking for bundlers and unit test runners separately feels better to me.